### PR TITLE
update QuickLRU option

### PR DESCRIPTION
**Pull request by canxuemianbao**

### CODE REVIEW

**Comment by canxuemianbao** [comment]
image
as [quickLRU](https://www.npmjs.com/package/@alloc/quick-lru) just mention they will set maxAge default to infinity, i think we can keep undefined acceptable

**Comment by StarpTech** [comment]
Hi @canxuemianbao please add a test.

**Comment by canxuemianbao** [comment]
@StarpTech updated

File: test/http-data-source.test.ts [hunk 1]
```diff
@@ -1336,6 +1336,56 @@ test('Should only cache GET successful responses with the correct status code',
   t.is(cacheMap.size, 0)
 })
 
+test('Response LRU cache will clear after 100ms when user set maxAge to 100ms', async (t) => {
```
Comment (StarpTech) [inline test/http-data-source.test.ts:1339]: I'd rename it to `Global maxAge should be used when no maxAge was set` or similar.

File: src/http-data-source.ts [hunk 2]
```diff
@@ -325,9 +326,8 @@ export abstract class HTTPDataSource<TContext = any> extends DataSource {
           })
           .catch((err) => this.logger?.error(err))
       }
-
       return response
-    } catch (error) {
+    } catch (error: any) {
```
Comment (StarpTech) [inline src/http-data-source.ts:330]: Why is it necessary?

File: test/http-data-source.test.ts (hunk 1 above)
Comment (canxuemianbao) [reply test/http-data-source.test.ts]: agree

File: src/http-data-source.ts (hunk 2 above)
Comment (canxuemianbao) [reply src/http-data-source.ts]: my typescript remains me should set type for error, and my version is v4.4.2, but we can keep it empty in lower version.

Comment (StarpTech) [reply src/http-data-source.ts]: I'm fine with that.

**Comment by StarpTech** [comment]
Could you fix the type issues?

**Comment by shkaper** [comment]
Sorry, I wasn't sure how active this is, so I've added #21 with a fix for the unit test. I'm preserving @canxuemianbao 's authorship on the commits there

**Comment by StarpTech** [comment]
This PR can be closed then?

**Comment by shkaper** [comment]
Yes. Please review the other one when you get a chance.

**Comment by StarpTech** [comment]
Will do.

**Pull request closed without merging**